### PR TITLE
fix mail adress read with globale settings set

### PR DESCRIPTION
**Pull request by iszmais**
https://mantis.ilias.de/view.php?id=36590

Fixes a bug caused in https://github.com/ILIAS-eLearning/ILIAS/pull/4881 where the user mail adress was not read properly when the user mail settings where disabled.

### CODE REVIEW

**Comment by iszmais** [comment]
@gvollbach This would normaly be part of @mjansenDatabay maintainance, but since he isnt available and this is quite urgentcould you take a look for review?

The changes a quite small therefore i assume this should be handable.

Greetings,
@iszmais

File: Services/Mail/classes/class.ilMailOptions.php [hunk 1]
```diff
@@ -142,6 +141,7 @@ protected function read() : void
             )) {
                 $this->emailAddressMode = self::FIRST_EMAIL;
             }
+            }
```
Comment (mjansenDatabay) [inline Services/Mail/classes/class.ilMailOptions.php:144]: This looks like wrong indentation:).

**Comment by mjansenDatabay** [comment]
LGTM! Please also pick this to `release_8` and `trunk`.

**Review by mjansenDatabay — APPROVED** [review]

**Comment by mjansenDatabay** [comment]
The tests still fail, this could be merged after these have been fixed.

**Comment by iszmais** [comment]
Thx for the feedback @mjansenDatabay !

I had to make some major changes on the test (and fixed a untracked bug alongside) therefore i fully refactored the tests and would appreciate if you could rereview those.

Greetings,
@iszmais

**Comment by iszmais** [comment]
Thx for the feedback @mjansenDatabay !

I had to make some major changes on the test (and fixed a untracked bug alongside) therefore i fully refactored the tests and would appreciate if you could rereview those.

Greetings,
@iszmais

**Comment by iszmais** [comment]
Thx for the feedback @mjansenDatabay !

I had to make some major changes on the test (and fixed a untracked bug alongside) therefore i fully refactored the tests and would appreciate if you could rereview those.

Greetings,
@iszmais

**Pull request merged**